crates/sui-adapter/src/adapter.rs

```diff
@@ -527,6 +530,10 @@ fn process_successful_execution<
                     &mut frozen_object_ids,
                     &mut child_count_deltas,
                 )?;
+                debug_assert!(match obj.owner {
```

Comment:
A rule of thumb: if you're checking something safety critical, like this, make it `assert`. It's much better to crash a validator than to let something unsafe happen.

Reply:
It's more of an annotation of an assumption of the post condition for `handle_transfer` (that the `handle_transfer` function is setting this field). If this ever changes, just about any test with shared objects will break, hence its just under `debug`

Reply:
The gas underflow bug we had a few weeks ago was allowed to happen by a `debug_assert!` that should have been an `assert!`, so I've become quite leary of `debug_assert!` in the critical path.

If this is just documenting an invariant that can't fail, there is no harm (other than perf) in making it a full assert, and it looks like a very cheap check. If we think its unnecessary because of test coverage, we could delete it or turn it into a comment.

Reply:
(I'm willing to change my mind here if there's an argument for why `assert!` would be harmful rather than merely unnecessary)